tests/jni_api.rs

```diff
@@ -410,6 +410,108 @@ pub fn get_byte_array_elements_auto() {
     assert_eq!(res[2], 4);
 }
 
+#[test]
+pub fn get_auto_long_array_elements() {
+    let env = attach_current_thread();
+
+    // Create original Java array
+    let buf: &[i64] = &[1, 2, 3];
+    let java_array = env
+        .new_long_array(3)
+        .expect("JNIEnv#new_long_array must create a java array with given size");
+
+    // Insert array elements
+    let _ = env.set_long_array_region(java_array, 0, buf);
+
+    // Use a scope to test Drop
+    {
+        // Get long array elements auto wrapper
+        let auto_ptr = env
+            .get_auto_long_array_elements(java_array, ReleaseMode::CopyBack)
+            .unwrap();
+
+        // Check pointer access
+        let ptr = auto_ptr.as_ptr();
+        assert_eq!(unsafe { *ptr.offset(0) }, 1);
+        assert_eq!(unsafe { *ptr.offset(1) }, 2);
+        assert_eq!(unsafe { *ptr.offset(2) }, 3);
+
+        // Check pointer From access
+        let ptr: *mut jlong = std::convert::From::from(&auto_ptr);
+        assert_eq!(unsafe { *ptr.offset(0) }, 1);
+        assert_eq!(unsafe { *ptr.offset(1) }, 2);
+        assert_eq!(unsafe { *ptr.offset(2) }, 3);
+
+        // Check pointer into() access
+        let ptr: *mut jlong = (&auto_ptr).into();
+        assert_eq!(unsafe { *ptr.offset(0) }, 1);
+        assert_eq!(unsafe { *ptr.offset(1) }, 2);
+        assert_eq!(unsafe { *ptr.offset(2) }, 3);
+
+        // Modify
+        unsafe {
+            *ptr.offset(0) += 1;
+            *ptr.offset(1) += 1;
+            *ptr.offset(2) += 1;
+        }
+
+        // Commit would be necessary here, if there were no closure
+        //auto_ptr.commit();
+    }
+
+    // Confirm modification of original Java array
+    let mut res: [i64; 3] = [0; 3];
+    env.get_long_array_region(java_array, 0, &mut res).unwrap();
+    assert_eq!(res[0], 2);
+    assert_eq!(res[1], 3);
+    assert_eq!(res[2], 4);
+}
+
+#[test]
+pub fn get_auto_long_array_elements_commit() {
+    let env = attach_current_thread();
+
+    // Create original Java array
+    let buf: &[i64] = &[1, 2, 3];
+    let java_array = env
+        .new_long_array(3)
+        .expect("JNIEnv#new_long_array must create a java array with given size");
+
+    // Insert array elements
+    let _ = env.set_long_array_region(java_array, 0, buf);
+
+    // Get long array elements auto wrapper
+    let mut auto_ptr = env
+        .get_auto_long_array_elements(java_array, ReleaseMode::CopyBack)
+        .unwrap();
+
```

Comment:
As the array may be a direct pointer (according to the spec, we don't know of VM vendor/version/GC combinations where it happens), I think we shall check if it is a copy, and if not — return early (or, if Rust has an [`Assumptions`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assumptions.html) alternative, use them).

Reply:
Makes sense. A quick search didn't show any `assume`, I added a simple condition and return early if it's not a copy.